numpy/tests/test_reloading.py

```diff
@@ -29,3 +31,30 @@ def test_novalue():
         assert_equal(repr(np._NoValue), '<no value>')
         assert_(pickle.loads(pickle.dumps(np._NoValue,
                                           protocol=proto)) is np._NoValue)
+
+
+def try_full_reload():
+    # Helper function for below test.
+    import sys
+    import numpy as np
+
+    for k in list(sys.modules.keys()):
+        if "numpy" in k:
+            del sys.modules[k]
+
+    import numpy as np
+
+
+def test_full_reimport():
+    """At the time of writing this, it is *not* truly supported, but
+    apparently enough users rely on it, for it to be an annoying change
+    when it started failing previously.
+    """
+    # Test within a subprocess, to ensure that we do not mess with the
+    # global state during the test run (could lead to cryptic test failures).
+    # This is generally unsafe, especially, since we also reload the C-modules.
+    p = Process(target=try_full_reload)
+    p.start()
+    p.join()
+    assert p.exitcode == 0
```

Comment:
Hmm, this fails on azure/windows with `FileNotFoundError: [Errno 2] No such file or directory: 'D:\\a\\1\\s\\build\\test\\runtests.py'` Just skip on windows:

https://dev.azure.com/numpy/numpy/_build/results?buildId=9919&view=logs&j=25bb66cf-4a16-533e-490c-8da4b5a3ec04&t=97446ee6-ebce-5662-24e5-f82eed312005&l=217

Reply:
other option is maybe try  https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods muliprocessing.set_start_method('spawn')

Reply:
not sure if this is already the default on windows for the version we are using

Reply:
Maybe 
```
txt = r"""
    import sys
    import numpy as np

    for k in list(sys.modules.keys()):
        if "numpy" in k:
            del sys.modules[k]

    import numpy as np
"""
p = subprocess.run([sys.executable, '-c', textwrap.dedent(txt)])
```

Reply:
Thanks matti, will just use that, seems simplest.

I changed the docs to link `ndarray.attribute`, did not realize the `~` would remove the ndarray as well.